superset-frontend/src/dashboard/util/findParentId.test.ts

```diff
@@ -41,4 +41,15 @@ describe('findParentId', () => {
   it('should return null if the parent cannot be found', () => {
     expect(findParentId({ childId: 'a', layout })).toBeNull();
   });
+
+  it('should not throw error and return null with bad / missing inputs', () => {
+    // @ts-ignore
+    expect(findParentId(null)).toBeNull();
+    // @ts-ignore
+    expect(findParentId({ layout })).toBeNull();
+    // @ts-ignore
+    expect(findParentId({ childId: 'a' })).toBeNull();
+    // @ts-ignore
```

Comment:
how difficult would it be to fix these ignores?

Reply:
@eschutho those are intentional because I am passing invalid data in the test file on purpose to simulate bad data creeping in somewhere in process of calling this function.  So if I change the tests to not have type errors, I cannot test bad inputs.  Because we have a mix of ts and regular js code, and we don't have consistent validation of API data and data mutations through redux and components, the TypeScript I added won't solve the issue where the function call still get called with bad data.  Does this make sense?

Reply:
Yeah, sounds good, thanks!